tests/datasets/test_sentinel.py

```diff
@@ -149,3 +155,36 @@ def test_invalid_query(self, dataset: Sentinel2) -> None:
 
     def test_float_res(self, dataset: Sentinel2) -> None:
         Sentinel2(dataset.paths, res=10.0, bands=dataset.bands)
+
+    @pytest.mark.parametrize('dataset_spec', [DatasetReader, WarpedVRT])
+    def test_footprint_from_datasource_metadata_file(
+        self, dataset: Sentinel2, dataset_spec: type[DatasetReader] | type[WarpedVRT]
+    ) -> None:
+        # WKT in EPSG:4326
+        metadata_footprint_wkt = dataset.index.geometry.to_crs(4326).values[0].wkt
+        footprint_wgs84 = shapely.wkt.loads(metadata_footprint_wkt)
+
+        # Fake dataset
```

Comment:
Can we test with real data instead? Just update the `tests/data/sentinel2/data.py` file to save the metadata you need.